packages/api-reference/src/features/Operation/Operation.vue

```diff
@@ -70,6 +73,29 @@ const selectedSecuritySchemes = computed(() =>
     securitySchemes,
   ).map(convertSecurityScheme),
 )
+
+/**
+ * Determine the effective server for examples/snippets.
+ * Preference order: operation.servers -> pathItem.servers -> provided server prop
+ */
+const effectiveServer = computed<ServerObject | undefined>(() => {
+  const opServer =
+    operation.value?.servers && operation.value.servers[0]
+      ? (getResolvedRef(operation.value.servers[0]) as ServerObject)
+      : undefined
+
+  if (opServer?.url) return opServer
+
+  const pathServer =
+    pathItem.value?.servers && pathItem.value.servers[0]
+      ? (getResolvedRef(pathItem.value.servers[0]) as ServerObject)
+      : undefined
+
+  if (pathServer?.url) return pathServer
+
+  // Fallback to the server coming from the active collection selection
+  return server as unknown as ServerObject | undefined
+})
```

Comment:
I think we’d want to put such logic in a separate file, with a dedicated test file, so we can test more (edge) cases easily.

Reply:
yep just this, move to a helper with this logic, add some tests and this should be good! Nice work